### PR TITLE
Add log and fix bug

### PR DESCRIPTION
**Pull request by shenli**
1. Add some logs
2. Fix a bug when checking charset
3. Update tidb in vendor
@siddontang @tiancaiamao @qiuyesuifeng PTAL

### CODE REVIEW

File: checker/checker.go [hunk 1]
```diff
@@ -88,8 +88,10 @@ func (c *checker) getTables() error {
 }
 
 func (c *checker) getCreateTable(tn string) (string, error) {
-	rs, err := querySQL(c.db, fmt.Sprintf("show create table %s;", tn))
+	stmt := fmt.Sprintf("show create table %s;", tn)
+	rs, err := querySQL(c.db, stmt)
 	if err != nil {
+		log.Errorf("Get create table failed: %s", stmt)
```
Comment (qiuyesuifeng) [inline checker/checker.go]: Should we log error information?

File: checker/checker.go [hunk 2]
```diff
@@ -109,17 +111,24 @@ func (c *checker) checkTable(tableName string) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
-	return c.checkCreateSQL(createSQL)
+	err = c.checkCreateSQL(createSQL)
+	if err != nil {
+		log.Errorf("Check create table failed: %s", createSQL)
```
Comment (qiuyesuifeng) [inline checker/checker.go]: Ditto.

File: checker/checker.go (hunk 1 above)
Comment (shenli) [inline checker/checker.go]: The error is logged at https://github.com/pingcap/tidb-tools/blob/master/checker/checker.go#L67

**Comment by shenli** [comment]
@qiuyesuifeng PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Comment by shenli** [comment]
@qiuyesuifeng @siddontang PTAL

File: checker/checker.go [hunk 3]
```diff
@@ -109,17 +109,22 @@ func (c *checker) checkTable(tableName string) error {
...
 func (c *checker) checkCreateSQL(createSQL string) error {
 	stmt, err := parser.New().ParseOneStmt(createSQL, "", "")
 	if err != nil {
-		return errors.Trace(err)
+		return errors.Trace(errors.Annotatef(err, " parse error %s", createSQL))
 	}
 	// Analyze ast
 	err = c.checkAST(stmt)
 	if err != nil {
+		log.Errorf("checkAST error: %s", err)
```
Comment (siddontang) [inline checker/checker.go]: need log here?

File: checker/checker.go [hunk 4]
```diff
@@ -109,17 +109,22 @@ func (c *checker) checkTable(tableName string) error {
...
+		return errors.Trace(err)
+	}
+	return nil
 }
 
 func (c *checker) checkCreateSQL(createSQL string) error {
 	stmt, err := parser.New().ParseOneStmt(createSQL, "", "")
 	if err != nil {
-		return errors.Trace(err)
+		return errors.Trace(errors.Annotatef(err, " parse error %s", createSQL))
```
Comment (siddontang) [inline checker/checker.go]: "parse %s error"

File: checker/checker.go (hunk 3 above)
Comment (shenli) [inline checker/checker.go]: Yes we need the log.

**Comment by shenli** [comment]
@siddontang PTAL

**Pull request merged**